src/main/java/com/sap/cloudfoundry/client/facade/adapters/LogCacheClient.java

```diff
@@ -103,18 +89,37 @@ private URI buildGetLogsUrl(UUID applicationGuid, LocalDateTime offset) {
                                    .toUri();
     }
 
-    private String parseBodyToString(InputStream is) {
-        try (InputStream wrapped = is) {
-            return IOUtils.toString(wrapped, StandardCharsets.UTF_8);
-        } catch (IOException e) {
-            throw new CloudException(String.format(Messages.CANT_READ_APP_LOGS_RESPONSE, e.getMessage()), e);
+    private InputStream getResponseBodyAsStream(URI logsUrl) {
+        List<InputStream> responseStreams = WEB_CLIENT.get()
+                                                      .uri(logsUrl)
+                                                      .headers(httpHeaders -> httpHeaders.addAll(getAdditionalRequestHeaders(requestTags)))
+                                                      .header(HttpHeaders.AUTHORIZATION, oAuthClient.getAuthorizationHeaderValue())
+                                                      .exchangeToFlux(this::handleClientResponse)
+                                                      .map(DataBuffer::asInputStream)
+                                                      .collectList()
+                                                      .block();
+        return new SequenceInputStream(Collections.enumeration(responseStreams));
```

Comment:
Are these streams closed anywhere? We should be careful about a resource leak

Reply:
Removed in the latest change. When closing the SequenceInputStream, it closes the streams that it is comprised of, at least according to doc:
_If this stream was created from an enumeration, all remaining elements are requested from the enumeration and closed before the close method returns._